### ISSUE TITLE
Convolutional LSTM raises errors

### ISSUE THREAD

**Issue by Junchi-Liang** [tensorflow/tensorflow#22946]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 18.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
No
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.9
- **Python version**:
2.7
- **Bazel version (if compiling from source)**:
N/A
- **GCC/Compiler version (if compiling from source)**:
N/A
- **CUDA/cuDNN version**:
CUDA 9
cuDNN 7.1

- **GPU model and memory**:
GeForce GTX 1070
8GB

- **Exact command to reproduce**:
Listed in source code

### Describe the problem

Using convolutional LSTM cell with dynamic_rnn raises errors:
```
2018-10-12 14:24:01.651741: E tensorflow/core/grappler/optimizers/dependency_optimizer.cc:581] Iteration = 0, topological sort failed with message: The graph couldn't be sorted in topological order.
2018-10-12 14:24:01.652168: E tensorflow/core/grappler/optimizers/dependency_optimizer.cc:581] Iteration = 1, topological sort failed with message: The graph couldn't be sorted in topological order.
```
The Code is listed below, it works on tf1.8. But tf1.9 and tf1.10 give errors above.

### Source code / logs

```
import tensorflow as tf
cell = tf.contrib.rnn.ConvLSTMCell(conv_ndims=2, input_shape=[28, 28, 3], output_channels=32, kernel_shape=[3, 3], use_bias=True)
c_input = tf.placeholder(tf.float32, [None] + list(cell.state_size.c))
h_input = tf.placeholder(tf.float32, [None] + list(cell.state_size.h))
state_in = tf.nn.rnn_cell.LSTMStateTuple(c_input, h_input)
x = tf.placeholder(tf.float32, [None, None, 28, 28, 3])
lstm_output, next_state = tf.nn.dynamic_rnn(cell = cell, inputs = x, initial_state = state_in, dtype = tf.float32, time_major = False)
import numpy as np
v_x = np.ones([1, 1, 28, 28, 3])
c_v = np.ones([1, 28, 28, 32])
h_v = np.ones([1, 28, 28, 32])
feed_dict = {x: v_x, c_input: c_v, h_input: h_v}
init_op = tf.variables_initializer(tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES))
sess = tf.Session()
sess.run(init_op)
test = sess.run(lstm_output, feed_dict = feed_dict)
```

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by fengyang0317** [comment]
I am facing the same issue. The coding is still able to run with this "topological sort failed" warning. I am wondering whether I will get correct training results with this warning.

In addition, static_rnn is ok with ConvLSTMCell.

**Comment by Junchi-Liang** [comment]
I haven't tested it, but I wonder if we implement a convolutional LSTM cell


> I am facing the same issue. The coding is still able to run with this "topological sort failed" warning. I am wondering whether I will get correct training results with this warning.
> 
> In addition, static_rnn is ok with ConvLSTMCell.
> 
> Another walkaround is to use extract_image_patches to implement convolution.

**Comment by yongtang** [comment]
@Junchi-Liang I run through the code with nightly build and it works fine. Could you give nightly a try and see if this is still an issue?

**Comment by Junchi-Liang** [comment]
I tried '1.12.0-dev20181012', but it cannot work as it still gives the same error.

@yongtang Could you tell me which version did you use?

**Comment by yongtang** [comment]
@Junchi-Liang I run with the following (not with GPU):
```
root@ubuntu:/v# python -c 'import tensorflow as tf; print(tf.VERSION)'
1.12.0-dev20181012
root@ubuntu:/v# python 22946.py       
2018-10-16 16:26:34.662982: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
root@ubuntu:/v# cat 22946.py 
import tensorflow as tf
cell = tf.contrib.rnn.ConvLSTMCell(conv_ndims=2, input_shape=[28, 28, 3], output_channels=32, kernel_shape=[3, 3], use_bias=True)
c_input = tf.placeholder(tf.float32, [None] + list(cell.state_size.c))
h_input = tf.placeholder(tf.float32, [None] + list(cell.state_size.h))
state_in = tf.nn.rnn_cell.LSTMStateTuple(c_input, h_input)
x = tf.placeholder(tf.float32, [None, None, 28, 28, 3])
lstm_output, next_state = tf.nn.dynamic_rnn(cell = cell, inputs = x, initial_state = state_in, dtype = tf.float32, time_major = False)
import numpy as np
v_x = np.ones([1, 1, 28, 28, 3])
c_v = np.ones([1, 28, 28, 32])
h_v = np.ones([1, 28, 28, 32])
feed_dict = {x: v_x, c_input: c_v, h_input: h_v}
init_op = tf.variables_initializer(tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES))
sess = tf.Session()
sess.run(init_op)
test = sess.run(lstm_output, feed_dict = feed_dict)
root@ubuntu:/v# 
```

**Comment by Junchi-Liang** [comment]
@yongtang @wt-huang 
I tried both 1.12.0-dev20181012 and 1.12.0.rc without GPU support, the code works. But it cannot work in their GPU versions. Does it imply a problem with CUDA or cuDNN? For instance, incompatible version.

**Comment by drewmccal** [comment]
I'm also seeing this issue on TensorFlow 1.10, with a custom estimator.

**Comment by LaurencePeanuts** [comment]
I'm also seeing this issue on TensorFlow 1.10.1 on a gpu.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by PharrellWANG** [comment]
Same issue here with TensorFlow version: 1.12.0, compiled from source with CUDA10

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ymodak** [comment]
Closing this issue since it no longer persists with the latest TF build. Thanks!

**Comment by Honghe** [comment]
Same issue here with TensorFlow-GPU: 1.12.0, with CUDA-9.
Code: https://github.com/Honghe/Speaker_Verification

**Comment by Honghe** [comment]
@ymodak 
> Closing this issue since it no longer persists with the latest TF build

Which TensorFlow version? I met this error in v1.12.0-GPU. Thanks

**Comment by Corea** [comment]
I also have this problem with Python 3.6.8, TensorFlow v1.12.2-0-gcf74798993, CUDA 9.0, cuDNN 7.5.0.
